### PR TITLE
Add Multiset

### LINKED ISSUE #390: Add `multiset` to `pydatastructs.miscellaneous_data_structures`
#### Description of the problem

https://github.com/codezonediitj/pydatastructs/issues/390#issuecomment-986180523

> We do have SelfBalancingBinaryTree we just need to put pop method (which is basically show top and delete). Apply pop enough time and we have tree sort. My point being that we don't need to reinvent anything.

### PR DESCRIPTION
**Pull request by sak-codes**
Fixes #390

### CODE REVIEW

File: pydatastructs/miscellaneous_data_structures/multiset.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+__all__ = [
+    'Multiset'
+]
+
+
+class Multiset:
+    def __init__(self, *args):
+        self.counter = dict()
```
Comment (czgdp1807) [inline pydatastructs/miscellaneous_data_structures/multiset.py:8]: Add a "TODO: Implement `dict` in `pydatastructs`".

**Review by czgdp1807 — APPROVED** [review]
Other than the review, rest LGTM.

File: pydatastructs/miscellaneous_data_structures/multiset.py (hunk 1 above)
Comment (sak-codes) [reply pydatastructs/miscellaneous_data_structures/multiset.py:9]: Done

**Review by czgdp1807 — APPROVED** [review]
Looks good.

**Pull request merged**